core/src/main/kotlin/org/evomaster/core/search/algorithms/MosaAlgorithm.kt

```diff
@@ -32,51 +42,73 @@ class MosaAlgorithm<T> : SearchAlgorithm<T>() where T : Individual {
 
             //new generation
 
-            val nextPop: MutableList<EvaluatedIndividual<T>> = mutableListOf()
+            val nextPop: MutableList<Data> = mutableListOf()
+
+            while (nextPop.size < n-1) {
 
-            while (nextPop.size < n) {
+                var ind = selection()
 
-                getMutatator().mutateAndSave(selection(), archive)
-                        ?.let { nextPop.add(it) }
```

Comment:
evaluating an individual might return null (long story...). so, should add to population only when != null. To do this, a nice way is Kotlin is to use "?.let" , which means: execute the lambda expression only if the caller is not null. because it is a "let", inside the lambda the "it" refers to the caller

Reply:
Given the new version of the code, I guess I should change these lines:
_var eval = getMutatator().mutateAndSave(ind, archive)
nextPop.add(Data(eval as EvaluatedIndividual))_
as follows:
getMutatator().mutateAndSave(ind, archive)?.let(nextPop.add(Data(it))_

Let me make this change and test it briefly on my laptop.